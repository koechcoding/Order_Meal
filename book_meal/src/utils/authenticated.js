export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}