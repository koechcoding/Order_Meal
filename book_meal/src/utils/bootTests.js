import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
