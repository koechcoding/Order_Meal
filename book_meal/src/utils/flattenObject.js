export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}