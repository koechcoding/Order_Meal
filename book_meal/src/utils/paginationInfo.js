export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}