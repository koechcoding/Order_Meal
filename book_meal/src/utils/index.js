import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}