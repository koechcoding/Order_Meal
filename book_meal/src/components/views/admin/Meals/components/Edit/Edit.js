import React from 'react';
import PropTypes from 'prop-types';
import axios from 'src/axios';
import Modal from 'src/components/common/Modal';
import ImageInput from 'src/components/common/ImageInput';
import { Alert, Input, Button } from 'reactstrap';
import { singleError } from 'src/utils';
import { IMAGES_UPLOAD_URL, IMAGE_UPLOAD_PRESET } from 'src/constants';

class EditModal extends React.Component {

    constructor(props){
        super(props);
        this.state = {
            ...props.meal
        }
    }

    onImageAdded=(data)=>{
        this.setState({
            ...this.state,
            image: data,
        });
    }

    onImageRemoved=()=>{
        this.setState({
            ...this.state,
            image: null,
        });
    }

    onPrefillRemoved=()=>{
        this.setState({
            ...this.state,
            image: null,
            img_url: "#",
        });
    }

    onOpened=()=>{
        this.setState({
            ...this.state,
            error: null,
            success: false,
            image: null,
        });
    }
}