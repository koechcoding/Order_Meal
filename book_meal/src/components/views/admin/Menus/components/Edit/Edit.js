import React from 'react';
import PropTypes from 'prop-types';
import axios from 'src/axios';
import Modal from 'src/components/common/Modal';
import { Alert, Button, Input } from 'reactstrap';
import { singleError } from 'src/utils';
import { Async } from 'react-select';
import _ from 'lodash';

import 'react-select/dist/react-select.css';
import './styles.css';

class EditModal extends React.Component {
    constructor(props){
        super(props);
        this.state={
            quantity: 1,
        };
    }

    reset=()=>{
        setTimeout(()=>{
            this.setState({
                ...this.state,
                success: false,
                error: null
            });
        }, 2000);
    }

    onOpened=()=>{
        this.setState({
            ...this.state,
            error: null,
            success: false,
            menus: [],
            meals: [],
        });

        if(this.props){
            const { menuItem ={} } = this.props;
            let defaultValues = {
                meal: null,
                menu: null,
            }
            if(menuItem['meal.id']){
                defaultValues.meal ={
                    value: menuItem['meal.id'],
                    label: menuItem['meal.name']
                }
            }
            if (menuItem['menu.id']) {
                defaultValues.menu = {
                    value: menuItem['menu.id'], 
                    label: menuItem['menu.name']
                }
            }


            this.setState({
                ...this.state,
                ...defaultValues,
                quantity: menuItem.quantity,
            });
        }
    }

    onChange = (e) => {
        this.setState({
            ...this.state,
            [e.target.name]: e.target.value
        })
    }

    fetchMenus = (input, callback) => {
        axios.get(`/menus?search=${input}&per_page=5`).then(({ data }) => {
            const menus = data.menus.map((menu) => {
                return { value: menu.id, label: menu.name }
            });
            callback(null, { options: menus });
        }).catch(({ response }) => {
            this.setState({
                ...this.state,
                error: response
            });
            callback(response, null);
        });
    }

    fetchMeals = (input, callback)=>{
        axios.get('/meals?search=name:${input}&per_page=5').then(({ data}) =>{
            const meals = data.meals.map((meal) => {
                return { value: meal.id, label: meal.name }
            });
            callback(null, { options: male});
        }).catch(({ response })=>{
            this.setState({
                ...this.state,
                error: response,
            });
            callback(response, null);
        });
    }

    setSelectedMenu= (meal) =>{
        this.setState({
            ...this.state,
            meal
        })
    }

    onEdit = () => {
        this.props.setLoading(true);
        const { meal = {}, menu = {} } = this.state;
        const menuItem = {
            meal_id: meal.value ? meal.value : '',
            menu_id: menu.value ? menu.value : '',
            quantity: this.state.quantity,
        };

        axios.put(`/menu-items/${this.props.menuItem.id}`, menuItem).then(({ data }) => {
            this.props.onChange();
            this.setState({
                ...this.state,
                success: true,
                error: null,
            });
            this.reset();
            this.props.setLoading(false);
            setTimeout(this.props.toggle, 1000);
        }).catch(({ response }) => {
            this.setState({
                ...this.state,
                error: response,
                success: false,
            });
            this.reset();
            this.props.setLoading(false);
        });
    }

    render(){
        const {isOpen, toggle } = this.props;
        const { error, success, meal, menu} = this.state;
        const body =(
            <div>
                {success &&
                  <Alert className="text-center text-small" color="success">
                      Successfully updated.
                  </Alert>
                }
                
                {error &&
                    <Alert className="text-center text-small" color="danger">
                        { singleError(error)}
                    </Alert>
                }
                <div className='pl-4 pr-4'>
                    <label>Meal Name</label>
                    <Async
                       name="meal"
                       value={meal}
                       onChange={this.setSelectedMeal}
                       loadOptions={_.throttle(this.fetchMeals, 500)}
                       filterOptions={(options) =>{ return options;}}
                    />
                    <label className='mt-3'>Select Menu</label>
                    <Async 
                       name="menu"
                       value={menu}
                       onChange={this.setSelectedMenu}
                       loadOptions={_.throttle(this.fetchMenus, 500)}
                       filterOptions={(options)=>{return options;}}
                    />
                    <label className='mt-3'>Quantity</label>
                    <input name="quantity" value={this.state.quantity} onChange={this.onChange} type="text" />
                </div>
            </div>
        );

        const footer = (
            <Button color="primary" className="m-auto" onClick={this.onEdit}>Update Item</Button>
        );
        return(
            <Modal 
               title="Edit Menu Item"
               body={body}
               footer={footer}
               isOpen={isOpen}
               toggle={toggle}
               onOpened={this.onOpened}
            />
        );
    }
}


EditModal.propTypes = {
    menuItem: PropTypes.object,
    toggle: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    setLoading: PropTypes.func.isRequired,
}

export default EditModal;