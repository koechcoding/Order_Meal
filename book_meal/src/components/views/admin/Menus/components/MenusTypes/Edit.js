
import React from 'react';
import PropTypes from 'prop-types';
import axios from 'src/axios';
import { Alert, Button, Input } from 'reactstrap';
import Modal from 'src/components/common/Modal';
import { singleError } from 'src/utils';

class Edit extends React.Component {

    state = { name: '' }

    onEdit = () => {
        this.props.setLoading(true);
        axios.put(`/menus/${this.props.menu.id}`, {name: this.state.name}).then(() => {
            this.props.setLoading(false);
            this.props.toggle();
            this.props.onChange();
        }).catch(({ response }) => {
            this.setState({
                ...this.state,
                error: response,
            });
            this.props.setLoading(false);
        });
    }

    onOpened = () => {
        this.setState({
            ...this.state,
            error: null
        })
    }

    onChange = (e) => {
        this.setState({
            ...this.state,
            [e.target.name]: e.target.value
        });
    }

    render() {
        const { error } = this.state;
        const { menu = {} } = this.props;
        const body = (
            <div>
                {error &&
                    <Alert className="text-center text-small" color="danger">
                        { singleError(error) }
                    </Alert>
                }
                <p>
                    <label>Name</label>
                    <Input 
                        name="name" 
                        defaultValue={menu.name} 
                        onChange={this.onChange}/>
                </p>
            </div>
        );

        const footer = (
            <Button color="primary" className="m-auto" onClick={this.onEdit}>Update</Button>
        );

        return (
            <Modal
                {...this.props}
                title="Edit Menu"
                body={body}
                footer={footer}
                toggle={this.props.toggle}
                onOpened={this.onOpened}
            />
        );
    }
}

Edit.propTypes = {
    toggle: PropTypes.func.isRequired,
    setLoading: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    menu: PropTypes.object.isRequired
}

export default Edit;