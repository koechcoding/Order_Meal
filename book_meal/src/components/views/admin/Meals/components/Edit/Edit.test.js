import React from 'react';
import { shallow, mount} from 'enzyme';
import 'src/utils/bootTests';
import Edit from './Edit';
import Adapter from 'enzyme-adapter-react=16';

const proops = { meal: {}, isOpen: true, toggle:()=>{}, onChange:()=>{}, setLoading:()=>{}}
describe('<Edit />', ()=>{
    it('renders correctly', ()=>{
        const wrapper = shallow(<Edit {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('adds image', ()=>{
        const wrapper = shallow(<Edit {...props} />);
        wrapper.instance().onImageAdded('test');
        expect(wrapper.state('image')).toBe('test');
    });
    it('resets on opened', ()=>{
        const wrapper = shallow(<Edit {...props} />);
        wrapper.setState({...wrapper.state(), error: true, success: true});
        wrapper.instance().onOpened();
        expect(wrapper.state('error')).toBe(null);
        expect(wrapper.state('success')).toBe(false);
    });
    it('removes image', () => {
        const wrapper = shallow(<Edit {...props} />);
        wrapper.instance().onImageRemoved();
        expect(wrapper.state('image')).toBe(null);
    });
    it('records input change', () => {
        const wrapper = mount(<Edit {...props} />);
        wrapper.instance().onChange({target: {name: 'name', value: 'value'}});
        expect(wrapper.state('name')).toBe('value');
    });
});