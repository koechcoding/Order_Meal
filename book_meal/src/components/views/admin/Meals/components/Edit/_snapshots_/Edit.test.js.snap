// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Edit /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditModal
    isOpen={true}
    meal={Object {}}
    onChange={[Function]}
    setLoading={[Function]}
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "body": <div>
        <ImageInput
                onImageAdded={[Function]}
                onImageRemoved={[Function]}
                onPrefillRemoved={[Function]}
                prefill={undefined}
        />
        <div
                className="pl-4 pr-4"
        >
                <label>
                         Name 
                </label>
                <Input
                        defaultValue={undefined}
                        name="name"
                        onChange={[Function]}
                        type="text"
                />
                <label>
                         Cost 
                </label>
                <Input
                        defaultValue={undefined}
                        name="cost"
                        onChange={[Function]}
                        type="number"
                />
        </div>
</div>,
      "footer": <Button
        className="m-auto"
        color="primary"
        onClick={[Function]}
        tag="button"
>
        Update Meal
</Button>,
      "isOpen": true,
      "onOpened": [Function],
      "title": "Edit Meal",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "body": <div>
          <ImageInput
                    onImageAdded={[Function]}
                    onImageRemoved={[Function]}
                    onPrefillRemoved={[Function]}
                    prefill={undefined}
          />
          <div
                    className="pl-4 pr-4"
          >
                    <label>
                               Name 
                    </label>
                    <Input
                              defaultValue={undefined}
                              name="name"
                              onChange={[Function]}
                              type="text"
                    />
                    <label>
                               Cost 
                    </label>
                    <Input
                              defaultValue={undefined}
                              name="cost"
                              onChange={[Function]}
                              type="number"
                    />
          </div>
</div>,
        "footer": <Button
          className="m-auto"
          color="primary"
          onClick={[Function]}
          tag="button"
>
          Update Meal
</Button>,
        "isOpen": true,
        "onOpened": [Function],
        "title": "Edit Meal",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;